### ISSUE TITLE
Are you still active in this repository?

### ISSUE THREAD

**Issue by talregev** [cowhi/lab_rl#1]
Are you still active in this repository? 

May I suggest fixes?

**Comment by cowhi** [comment]
not working actively on this but if you have some fixes, please go ahead and let me know :)

**Comment by talregev** [comment]
Okey some of the fixes:
1.  edit BUILD file you also need to add buffer.py file:

```
py_binary(
    name = "launcher",
    srcs = ["lab_rl/launcher.py",
            "lab_rl/agents.py",
            "lab_rl/environments.py",
            "lab_rl/experiments.py",
            "lab_rl/models.py",
            "lab_rl/helper.py",
```
 **`"lab_rl/buffer.py",`**
```
            "lab_rl/replaymemories.py"],
    data = [":deepmind_lab.so"],
    main = "lab_rl/launcher.py",
    )
```

2. You need copy files like that:
```
!cp -R lab/lab_rl/lab_rl/ lab/game_scripts/levels/
!cp -R lab/lab_rl/lab_rl/ lab/game_scripts/ 
!cp -R lab/lab_rl/lab_rl/*.map lab/assets/maps/src/
```
not sure about 2 first, that how it asking me, 
the last one it because they change the locations of .map files.

3. you need run: `bazel build :map_assets` and after that extract the pk3 files from cache bazel folder, and copy the bsp and aas files to `lab/assets/maps/built`. maybe it better to create the bsp and aas file and upload to your repository and tell copy them to `lab/assets/maps/built`.

4. on file [model.py][1] i change conv2d to conv3d.In tensorflow v1.12 the api was changed. I am not sure this is the neural network you mean on the new api (with conv3d).

5. on file [agents.py][2] i add this code, because after init the `self.model_name` stay None. I guess it deeper problem with argument that you can look for. 

```python
if self.model_name is None:
    self.model_name = "DQN_0000"
```
6. also change one of the files: [random_spawn_factory.lua][3], [random_spawn_factory_map.lua][4] because `random.seed(seed)` was not working. need more to investigate.

All these fix and I still didn't manage to run lab_rl.
I create a google collaborate with deepmind and lab_rl to test it.
You can use it too:
https://colab.research.google.com/drive/1P4l_iFbzRGdMsnnmyAlWhJjqLY7qIWWE

[1]:https://github.com/cowhi/lab_rl/blob/master/models.py#L75
[2]:https://github.com/cowhi/lab_rl/blob/master/agents.py#L475
[3]:https://github.com/cowhi/lab_rl/blob/master/lab_rl/random_spawn_factory.lua#L42
[4]:https://github.com/cowhi/lab_rl/blob/master/lab_rl/random_spawn_factory_map.lua#L43

**Comment by cowhi** [comment]
ok, thanks for those suggestions. I will take a closer look in the next couple of weeks. 

But one thing that I have noticed is that since I wrote that code they made quite some changes to the Deepmind Lab and I think it might have broken my code. Need to test it to see what is happening exactly.

At least they have their own description now on how to make your own maps. Maybe you can build on this?! Good luck!
https://github.com/deepmind/lab/blob/master/docs/developers/minimal_level_tutorial.md
